docs/en/reference/testing.rst

```diff
@@ -0,0 +1,74 @@
+Testing Guidelines
+===================
+
+In order to ensure high quality, all components the Doctrine DBAL library are extensively covered with tests.
+
+Having the code covered with tests and running all tests against each individual code change helps prevent
+breakages of the library logic when its code changes.
+
+Additionally, tests:
+
+1. Help maintainers understand what problem the given code change is trying to solve.
+2. Make sure that the problem being solved needs to be solved in the DBAL.
+3. Document the proper usage of the DBAL APIs.
+
+Requirements
+------------
+
+1. Each pull request that adds new or changes the existing logic must have tests.
```

Comment:
That sentence looks correct but sounds a bit awkward, I'd rather have some repetition here.
```suggestion
1. Each pull request that adds new logic or changes the existing logic must have tests.
```